pkg/machines/components/vmDisksTab.jsx

```diff
@@ -162,13 +162,27 @@ class VmDisksTab extends React.Component {
                 columns.push({ title: disk.diskExtras || '' });
 
             if (provider.name === 'LibvirtDBus') {
+                const onRemoveDisk = () => {
+                    return dispatch(detachDisk({ connectionName:vm.connectionName, id:vm.id, name:vm.name, target: disk.target, live: vm.state == 'running' }))
+                            .catch(ex => {
+                                onAddErrorNotification({
+                                    text: cockpit.format(_("Disk $0 fail to get detached from VM $1"), disk.target, vm.name),
+                                    detail: ex.message, resourceId: vm.id,
+                                });
+                            })
+                            .then(() => dispatch(getVm({ connectionName: vm.connectionName, id:vm.id })));
+                };
+
                 const diskActions = (
                     <div className='machines-listing-actions'>
-                        <RemoveDiskAction dispatch={dispatch}
-                            vm={vm}
-                            target={disk.target}
-                            idPrefixRow={idPrefixRow}
-                            onAddErrorNotification={onAddErrorNotification} />
+                        <DeleteResource objectType="Disk"
+                           className='machines-listing-actions'
+                           objectName={disk.target}
+                           objectId={vm.name + "-disk-" + disk.target}
+                           disabled={vm.state != 'shut off' && vm.state != 'running'}
+                           overlayText={_("The VM needs to be running or shut off to detach this device")}
+                           actionName={_("Remove")}
```

Comment:
Was this somehow intentional? The name of the action was deliberately called detach since the disk was not getting deleted.
Remove (at least from my understanding) somehow implies also the removal of the filesystem.

Reply:
There was quite some discussion about that in comments above. Designers in the end suggested "Remove" nevertheless.